testdata/project-v3-with-deploy-image/controllers/busybox_controller.go

```diff
@@ -187,6 +221,19 @@ func labelsForBusybox(name string) map[string]string {
 	return map[string]string{"type": "busybox", "busybox_cr": name}
 }
 
+func (r *BusyboxReconciler) finalizeBusyboxFinalizer(ctx context.Context, log logr.Logger, m *examplecomv1alpha1.Busybox) error {
```

Comment:
After we replace this code to create an event before remove the CR we also will need to add in the e2e test a check to ensure that the event was raised